java/arcs/core/storage/BackingStore.kt

```diff
@@ -53,7 +53,18 @@ class BackingStore<Data : CrdtData, Op : CrdtOperation, T>(
      */
     suspend fun getLocalData(referenceId: String) = store(referenceId).store.getLocalData()
 
-    suspend fun clearStoresCache() = storeMutex.withLock { stores.clear() }
+    /** Removes [DirectStore] caches and closes those that can be closed safely. */
+    fun clearStoresCache() {
+        while (!storeMutex.tryLock()) { /* Wait */ }
+        stores.forEach { _, sr ->
+            /** Aligns with the logic of [stores]'s [LruCacheMap.onEvict]. */
+            if (!sr.store.closed) {
+                sr.store.close()
```

Comment:
if it's possible for this call to throw, it might make sense to put the `unlock` below in a `finally` block.

Reply:
Good point!

Reply:
Done